alby/alby_oauth_service.go

```diff
@@ -1166,43 +1166,196 @@ func (svc *albyOAuthService) GetLSPChannelOffer(ctx context.Context) (*LSPChanne
 	return lspChannelOffer, nil
 }
 
+func (svc *albyOAuthService) GetLSPInfo(ctx context.Context, lsp, network string) (*LSPInfo, error) {
+	token, err := svc.fetchUserToken(ctx)
+	if err != nil {
+		logger.Logger.WithError(err).Error("Failed to fetch user token")
+		return nil, err
+	}
+
+	var client *http.Client
+	if token != nil {
+		client = svc.oauthConf.Client(ctx, token)
+		client.Timeout = 30 * time.Second
```

Comment:
`client.Timeout` could be set under the conditional, and then we don't need to set it twice